includes/legacy/class-wc-gateway-amazon-payments-advanced-legacy.php

```diff
@@ -1,4 +1,1850 @@
 <?php
-abstract class WC_Gateway_Amazon_Payments_Advanced_Legacy extends WC_Gateway_Amazon_Payments_Advanced_Abstract {
+class WC_Gateway_Amazon_Payments_Advanced_Legacy extends WC_Gateway_Amazon_Payments_Advanced_Abstract {
 
+	/**
+	 * Amazon Order Reference ID (when not in "login app" mode checkout)
+	 *
+	 * @var string
+	 */
+	protected $reference_id;
+
+	/**
+	 * Amazon Pay Access Token ("login app" mode checkout)
+	 *
+	 * @var string
+	 */
+	protected $access_token;
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		// Get Order Refererence ID and/or Access Token.
+		$this->reference_id = WC_Amazon_Payments_Advanced_API::get_reference_id();
```

Comment:
this two method call along with other in his file may be moved to legacy api class

Reply:
Yes, further cleanup may be done at the end of the whole implementation.